### PR TITLE
kwil-cli: Introduce broadcast tx commit timeout

### PR DESCRIPTION
**Pull request by charithabandi**

### CODE REVIEW

**Review by jchappelow — APPROVED** [review]

**Comment by jchappelow** [comment]
@charithabandi I overlooked the need to combine RPC error with broadcast error details to the timeout gets shown client side.  Just pushed a commit to fix this.

```
$  ./kwil-cli database deploy -p ~/test_db-nomath.kf --chain-id "" --sync
failed to deploy database: err code = 4, msg = timed out waiting for tx to be included in a block
TxHash: 595ca512ad9193c62e3c73105b1d7c565c19b25e205709e3afaae2a28a5cf3bc
```

The above was just showing `failed to deploy database: err code = 0, msg = ` since the result from check tx was fine and it was overwriting the timeout error.  They are now combined.

Here's with it mined, but with exec error, combining the messages and code from details:


```
 $  ./kwil-cli database deploy -p ~/test_db-nomath.kf --chain-id "" --sync
failed to deploy database: err code = 3, msg = broadcast error, detail: dataset exists: xc4503ccf15daada96e36cbbacf641f48ebc6226e28906c8757866b91 (120)
TxHash: 94f4d9cb36fa7e947e2e97fdd028c41469b747c2b9f7683e1d3c90bcb9003b9e
```

**Review by jchappelow — APPROVED** [review]
Working as intended.

**Comment by charithabandi** [comment]
> @charithabandi I overlooked the need to combine RPC error with broadcast error details to the timeout gets shown client side. Just pushed a commit to fix this.
> 
> ```
> $  ./kwil-cli database deploy -p ~/test_db-nomath.kf --chain-id "" --sync
> failed to deploy database: err code = 4, msg = timed out waiting for tx to be included in a block
> TxHash: 595ca512ad9193c62e3c73105b1d7c565c19b25e205709e3afaae2a28a5cf3bc
> ```
> 
> The above was just showing `failed to deploy database: err code = 0, msg = ` since the result from check tx was fine and it was overwriting the timeout error. They are now combined.
> 
> Here's with it mined, but with exec error, combining the messages and code from details:
> 
> ```
>  $  ./kwil-cli database deploy -p ~/test_db-nomath.kf --chain-id "" --sync
> failed to deploy database: err code = 3, msg = broadcast error, detail: dataset exists: xc4503ccf15daada96e36cbbacf641f48ebc6226e28906c8757866b91 (120)
> TxHash: 94f4d9cb36fa7e947e2e97fdd028c41469b747c2b9f7683e1d3c90bcb9003b9e
> ```

LGTM!

**Review by brennanjl — APPROVED** [review]

File: core/rpc/client/user/http/error.go [hunk 1]
```diff
@@ -95,12 +95,17 @@ func parseBroadcastError(respTxt []byte) (bool, error) {
...
-			// Reset the generic code and message in the RPCError with the
-			// broadcast-specific details. NOTE: this will overwrite if there
-			// are more than one details object, which is not expected.
-			rpcErr.Code = int32(txCode)
-			rpcErr.Msg = bcastErr.Message
-			if bcastErr.Hash != "" { // if there is a tx hash, include it (possibly just executed it)
+			// Supplement the message in the RPCError with the details, if there
+			// are any. NOTE: this will overwrite if there are more than one
+			// details object, which is not expected.
+			if bcastErr.Message != "" || txCode != transactions.CodeOk {
```
Comment (Yaiba) [inline core/rpc/client/user/http/error.go:101]: those error message manipulation really get hairy..

**Pull request merged**